generic3g/vertical/ModelVerticalGrid.F90

```diff
@@ -65,89 +63,99 @@ module function can_connect_to(this, src, rc)
 
 contains
 
-   function new_ModelVerticalGrid_basic(num_levels, units) result(vgrid)
+   function new_ModelVerticalGrid_basic(standard_name, units, num_levels) result(vgrid)
       type(ModelVerticalGrid) :: vgrid
-      integer, intent(in) :: num_levels
+      character(*), intent(in) :: standard_name
       character(*) , intent(in) :: units
-      !# character(*), intent(in) :: short_name
-      !# character(*), intent(in) :: standard_name
-      !# type(StateRegistry), pointer, intent(in) :: registry
+      integer, intent(in) :: num_levels
 
       call vgrid%set_id()
+      vgrid%standard_name = standard_name
       call vgrid%set_units(units)
       vgrid%num_levels = num_levels
-      !# vgrid%short_name = short_name
-      !# vgrid%standard_name = standard_name
-      !# vgrid%registry => registry
    end function new_ModelVerticalGrid_basic
 
    integer function get_num_levels(this) result(num_levels)
       class(ModelVerticalGrid), intent(in) :: this
       num_levels = this%num_levels
    end function get_num_levels
 
-   subroutine add_variant(this, short_name)
+   subroutine add_short_names(this, edge, center)
```

Comment:
Not sure that this is what we want here.  It is conceivable that someone could provide just the edge and not the center. or vice versa.     The center can in theory be derived from the edge (with some assumptions that appear to be nigh universal.)

It can stay for now, but be prepared for something where instead you specify name and staggerloc.